.github/workflows/CI.yml

```diff
@@ -1,19 +1,23 @@
 # This workflow will install Python dependencies, run tests and lint with a single version of Python
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
-name: CI 
+name: CI
 
 on:
-  [push]
-  
+  push:
+    tags:
+      - 'v*'  # Execute if version is tagged
+    branches:
+      - main  # Or if it is push to main
+  pull_request:  # Or if it is any pull_request
```

Comment:
I would rather see CI run on all pushes and pull request, also those updated ones. I saw an issue that a reviewed pull request breaks our code base, because CI is not run on updates.
Do you have some suggestions on restricting the scope?

Reply:
Do you have a link to the issue you saw so that others may look at it.

Reply:
@hsorby, this is an example: https://github.com/nih-sparc/sparc.client/pull/29#pullrequestreview-1760315190 
CI.yml pipeline wasn't run on updated pull request. As the result, we were not able to see code coverage.

Reply:
Closing